als-common/shared/src/main/scala/org/mulesoft/als/common/YPartBranch.scala

```diff
@@ -190,6 +192,8 @@ case class YPartBranch(node: YPart, position: AmfPosition, stack: Seq[YPart], is
 
 object NodeBranchBuilder {
   def getAstForRange(ast: YPart, startPosition: AmfPosition, endPosition: AmfPosition, isJson: Boolean): YPart = {
+
+    // TODO: new traverse check deletion ... could we compare start and endPosition before traverse both?
```

Comment:
Can we check this TODO? @llibarona 
(I think that is ok if we merge this and handled this change in another issue, but just to no forget about this)

Reply:
Tal cual. Si start == end, busco una sola vez no dos. El cache parece algo mas elaborado que un simple if que se puede agregar casi ciego, y de todas formas hay que manejar la primera vez que se traversa, cuando el cache esta vacio, para registrar una sola vez la posicion y no dos.
Igual lo veo mas simple, busco el startPosition, luego meto un if, si start == end, devuelvo solo el start si no busco tambien el end.